src/Parsers/ASTCreateUserQuery.h

```diff
@@ -51,6 +53,8 @@ class ASTCreateUserQuery : public IAST, public ASTQueryWithOnCluster
     std::shared_ptr<ASTSettingsProfileElements> settings;
     std::shared_ptr<ASTRolesOrUsersSet> grantees;
 
+    String default_database;
```

Comment:
I think this should be `std::optional<String> default_database`. Because `ASTCreateUserQuery` is used for `ALTER USER` statements too, and we need a way to clear once assigned default database for created user. 

I mean
1) `CREATE USER user1`
should work the same as
2) `CREATE USER user1 DEFAULT DATABASE ''`

but there should be difference between

3) `ALTER USER user1`
and
4) `ALTER USER user1 DEFAULT DATABASE ''`

The command `4` of course should remove the assignment for default database for `user1` class while the command `3` should not change it.

Reply:
I would be nice if you could support a new syntax `DEFAULT DATABASE NONE` as a synonym for `DEFAULT DATABASE ''`

Reply:
Thank you for helping me to carefully review, and this function is really necessary. If we want to add the syntax of `DEFAULT DATABASE NONE`, do we need to add some restrictions on the syntax of `CREATE DATABASE`, otherwise when users create a database named none,`DEFAULT DATABASE NONE` will be ambiguous.

Reply:
> I would be nice if you could support a new syntax `DEFAULT DATABASE NONE` as a synonym for `DEFAULT DATABASE ''`

Thank you for helping me to carefully review, and this function is really necessary. If we want to add the syntax of `DEFAULT DATABASE NONE`, do we need to add some restrictions on the syntax of `CREATE DATABASE`, otherwise when users create a database named none,`DEFAULT DATABASE NONE` will be ambiguous.

Reply:
No, I don't think we should put restrictions on `CREATE DATABASE` because of that. In the very  rare case when someone calls his own database as `NONE` he will be still able to assign it as default with backticks:
DEFAULT DATABASE \`NONE\`